server/thread/queries.ts

```diff
@@ -63,3 +64,48 @@ export const canUserSeeThread = async (options: CanUserSeeThreadOptions): Promis
 	const [maybeUserId] = await filterUsersWhoCanSeeThread({ threadId, userIds: [userId] });
 	return maybeUserId === userId;
 };
+
+type CreateThreadWithCommentOptions = {
+	text: string;
+	content: types.DocJson;
+	userId: null | string;
+	commenterName: null | string;
+};
+
+export const creatThreadCommentWithUserOrCommenter = async (
```

Comment:
This should be in `server/threadComment/queries.ts` since it creates a `ThreadComment`, and doesn't deal with `Thread` objects at all.